### PR TITLE
🎉 Remove hash when it is not necessary from normalization outputs

### LINKED ISSUE #2389: Check/validate host field when setting up snowflake
## Expected Behavior
_Tell us what should happen._
A user on slack created a pipeline sync between Google Sheet and Snowflake.
Unfortunately, the sync failed to complete but:

1. it managed to read data from the Google Sheet
1. it managed to write data to the `_airbyte_raw_sample` table
1. it managed to run the normalization script that generates the DBT SQL files
1. it failed to run DBT because the `host` field of the snowflake destination was `http://accountname.us-east-2.aws.snowflakecomputing.com/` instead of `accountname.us-east-2.aws.snowflakecomputing.com/`

## Current Behavior
_Tell us what happens instead of the expected behavior._
An exception was thrown by DBT-0.18.1:

```
50003: Failed to execute request: HTTPSConnectionPool(host='https', port=443): Max retries exceeded with url: //XXXXX.snowflakecomputing.com:443/session/v1/login-request?request_id=XXXX&databaseName=DEMO_DB&schemaName=PUBLIC&warehouse=COMPUTE_WH&roleName=SYSADMIN (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f2f1515da90>: Failed to establish a new connection: [Errno -2] Name or service not known'))
```

(note the `host='https'`)

Either 

* we need to upgrade to a DBT version that handles this sort of URL/host correctly 
* we need to make sure the host string does not start with `http://` so that DBT is not parsing the URL and thinking the host is http
* when generating the DBT profile.yml configuration file, we make sure the URL is well-formatted without that prefix

## Steps to Reproduce
1. Create a snowflake destination
1. copy / paste the host value including 'http://' at the beginning
1. run the sync with normalization

## Severity of the bug for you
Low 

## Airbyte Version
0.16.1-alpha

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368217717229) by [Unito](https://www.unito.io)

### LINKED ISSUE #3523: make normalization's  hash inclusion in unnested table names optional
The release of this change depends on https://github.com/airbytehq/airbyte/issues/3522

Consumers of tables produced by normalization cannot easily use nested tables due to the presence of hashes (used for collision prevention). These hashes are generally unnecessary outside of Postgres/similar dbs with low char limits for table names.

We should make this optional and set this to off by default except for Postgres/similar.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368129700092) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by ChristopheDuong**
## What
Closes #3523
Closes #2389

## How
Introduce a two-stage loop through the catalog

1. The first step is to collect all stream names (from top-level and nested ones) in the catalog
2. Then, resolve the naming issues by taking care of all naming from this global view of the catalog (not stream by stream) (hash are used in the names only if necessary)
3. Finally, actually generates model files using the resolved names.

In the future, we could separate the first stage when setting up the connection page, so the UI could have a chance to customize how to resolve naming conflicts. (The table name registry can then be imported from a file)

## Pre-merge Checklist
- [x] *Run integration tests*
- [x] *Publish Docker images*

## Recommended reading order
Ignore .sql and .json files changes

1. `airbyte-integrations/bases/base-normalization/normalization/transform_catalog/table_name_registry.py`
1. `airbyte-integrations/bases/base-normalization/normalization/transform_catalog/catalog_processor.py`
1. `airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py`
1. the rest

### CODE REVIEW

File: airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
 import pkgutil
 import shutil
 from enum import Enum
+from typing import Any, Dict
```
Comment (ChristopheDuong) [inline airbyte-integrations/bases/base-normalization/normalization/transform_config/transform.py:32]: Changes in this file are made to comply to MyPy...

**Review by davinchia — APPROVED** [review]
Nice job Chris! Makes lots of sense. Only one comment from me. 

I appreciated the great PR description and comments and the use of MyPy. Made much easier to review.

Feel free to merge whenever.

File: airbyte-integrations/bases/base-normalization/normalization/transform_catalog/catalog_processor.py [hunk 2]
```diff
@@ -63,34 +64,31 @@ def process(self, catalog_file: str, json_column_name: str, default_schema: str)
...
-        ):
+        )
+        for stream_processor in stream_processors:
             # Check properties
             if not stream_processor.properties:
                 raise EOFError("Invalid Catalog: Unexpected empty properties in catalog")
-
+            stream_processor.collect_table_names()
+        for conflict in tables_registry.resolve_names():
+            print(f"WARN: Resolving conflict: {conflict[0]}.{conflict[1]} from '{'.'.join(conflict[2])}' into {conflict[3]}")
```
[the remark this answers is not in the document]
Comment (ChristopheDuong) [reply airbyte-integrations/bases/base-normalization/normalization/transform_catalog/catalog_processor.py:86]: in the future, these "exceptions/conflicts" could be collected and displayed as warnings in the UI when setting up normalization for the connection

**Pull request merged**